Tests failed:

Inspecting 36 files
[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m

36 files inspected, [32mno offenses[0m detected

Tip: Based on detected gems, the following RuboCop extension libraries might be helpful:
  * rubocop-rake (https://rubygems.org/gems/rubocop-rake)
  * rubocop-rspec (https://rubygems.org/gems/rubocop-rspec)

You can opt out of this message by adding the following to your config (see https://docs.rubocop.org/rubocop/extensions.html#extension-suggestions for more options):
  AllCops:
    SuggestExtensions: false

Randomized with seed 38690

Gotsha::Actions::Test
  test
    with a test command configured
[32m      runs the command[0m
    without test command configured
[32m      fails with proper error[0m
    with only a blank (empty string) test command configured
[32m      fails with proper error[0m

Gotsha::Actions::Init
  init
    first time
[32m      creates default files and Git configuration[0m
    second time
[32m      does not create any files; onle Git configuration[0m

Gotsha::Actions::Fetch
  fetch
[32m    calls the Git command to fetch notes[0m

Gotsha::ActionDispatcher
  with a valid command followed by '-h'
[31m    calls `help` with the action (FAILED - 1)[0m
  with autogenerated config
git config core.hooksPath
.gotsha/hooks

[32m    fails with proper error[0m
  with a valid action
[32m    calls the action[0m
  without any action name
[32m    calls Help action[0m
  with unknown action name
[32m    raises HardFail error with user friendly message[0m
  with no config
    with `init` action
[32m      calls the action witohut exception[0m
    with other action than `init`
[32m      fails with proper error[0m
  with help action shortcut
[32m    calls the help action[0m
  with an extra unexpected argument
[32m    calls the action[0m

Gotsha::Actions::Status
  status
    when tests did not run (note with Git error message)
[32m      raises HardFail error[0m
    when tests failed
[32m      raises HardFail error[0m
    when tests ran fine
[32m      returns success message[0m
    when tests did not run (empty note)
[32m      raises HardFail error[0m

Gotsha::Actions::Help
  help
    without any argument
[32m      returns all public actions names[0m
    with a valid action name passed
[32m      returns the action description[0m
    with an unknown action name passed
[32m      raises exception with proper message[0m
    with a valid internal action name passed
[32m      returns the action description with 'internal command' warning[0m

Gotsha::Actions::Configure
  config
    without ENV['EDITOR'] set
[32m      fails with proper error[0m
    with ENV['EDITOR'] set to `vim`
[32m      issues command to open config file in the editor[0m

Gotsha::BashCommand
  .run!
    when verbose output disabled
[32m      returns collected stdout and successful status[0m
  .silent_run!
    when verbose output enabled
[32m      calls `.run!` with the same command[0m
    when verbose output disabled
[32m      calls `.run!` with output silencing[0m

Gotsha::Actions::Commit
  commit
[32m    creates a blank commit and triggers `test` on it[0m

Gotsha::Actions::Show
  show
    when last commit tests failed
[32m      returns not verified[0m
    when last commit was not verified
[32m      returns not verified[0m
    when last commit test were success
[32m      returns the result[0m

Gotsha::Actions::Uninstall
  uninstall
Removing config files...
Unsetting Git hooks path...
[32m    removes config files and Git configuration[0m

Gotsha::UserConfig
  when the config file does not exists
[32m    does not raise any exception[0m
  when config file exists
[32m    returns the content[0m

Gotsha::Actions::Push
  fetch
    when first push fails
First push detected, need to run tests one more time...
[32m      calls fetching[0m
    when first push works
[32m      ends there and does not fetch from remote[0m

Failures:

  1) Gotsha::ActionDispatcher with a valid command followed by '-h' calls `help` with the action
     [31mFailure/Error: return Actions::Help.new.call(action_name) if args == [HELP_ACTION_SHORTCUT][0m
     [31m[0m
     [31m  (#<Gotsha::Actions::Help:0x00007f2e9ab4e310>).call(#<Gotsha::Actions::Help:0x00007f2e9ab4e310>, "commit")[0m
     [31m      expected: 0 times with arguments: ("commit")[0m
     [31m      received: 1 time with arguments: (#<Gotsha::Actions::Help:0x00007f2e9ab4e310>, "commit")[0m
     [36m# ./lib/gotsha/action_dispatcher.rb:22:in 'Gotsha::ActionDispatcher#call'[0m
     [36m# ./lib/gotsha/action_dispatcher.rb:12:in 'Gotsha::ActionDispatcher.call'[0m
     [36m# ./spec/gotsha/action_dispatcher_spec.rb:102:in 'block (3 levels) in <top (required)>'[0m
     [36m# ------------------[0m
     [36m# --- Caused by: ---[0m
     [36m# ArgumentError:[0m
     [36m#   wrong number of arguments (given 1, expected 0)[0m
     [36m#   ./lib/gotsha/actions/commit.rb:8:in 'call'[0m

Finished in 0.37269 seconds (files took 0.46339 seconds to load)
[31m37 examples, 1 failure[0m

Failed examples:

[31mrspec ./spec/gotsha/action_dispatcher_spec.rb:99[0m [36m# Gotsha::ActionDispatcher with a valid command followed by '-h' calls `help` with the action[0m

Randomized with seed 38690
